app/src/main/java/org/eyeseetea/malariacare/domain/usecase/GetDeviceUseCase.java

```diff
@@ -0,0 +1,39 @@
+package org.eyeseetea.malariacare.domain.usecase;
+
+import org.eyeseetea.malariacare.domain.boundary.executors.IMainExecutor;
+import org.eyeseetea.malariacare.domain.boundary.repositories.IDeviceRepository;
+import org.eyeseetea.malariacare.domain.entity.Device;
+
+public class GetDeviceUseCase implements UseCase {
```

Comment:
use cases should use AsyncExecutor to retrieve data from data layer and MainExecutor to notify presentation layer

Reply:
But the device info is not in data layer, it's an Android service.

Reply:
well, we can do two things:
- use AsyncExecutor and MainExecutor as in other use cases for we have a standard way, this is better option for me.
- Does not use AsyncExecutor and MainExecutor because really here does not communicate with a database or remote service.
But you are using only main executor in your solution and this option has no sense